### PR TITLE
Make JAX ODE solver work with arbitrary-dimensional functions

### PR DESCRIPTION
**Pull request by romanngg**
Currently it only accepts vector-valued functions.

### CODE REVIEW

File: jax/experimental/ode.py [hunk 1]
```diff
@@ -448,7 +449,7 @@ def onearg_odeint(args):
   dim = 10
   t0 = 0.1
   t1 = 0.2
-  y0 = np.linspace(0.1, 0.9, dim)
+  y0 = np.linspace(0.1, 0.9, dim).reshape((5, 1, 2))
```
Comment (mattjj) [inline jax/experimental/ode.py]: Instead of changing existing tests, would it be possible to add new tests?

**Comment by mattjj** [comment]
Thanks!

I wonder: can a user make use of `vmap` to get this same behavior, without us having to change the odeint code itself?

File: jax/experimental/ode.py (hunk 1 above)
Comment (romanngg) [reply jax/experimental/ode.py]: Done! 95d20d24109d86f66f4465014fa040c422fcf05d

**Comment by romanngg** [comment]
Thanks Matt, yes, I can definitely flatten the function on my end, so feel free to ignore this PR. Added a test in case if you want to include it.

**Comment by hawkinsp** [comment]
@mattjj should we close this PR?

**Comment by mattjj** [comment]
@hawkinsp yes. I think we let it get stale, and odeint has grown this functionality now.

Thanks for pushing us in the right direction @romanngg!

**Pull request closed without merging**